extensions/guacamole-auth-jdbc/modules/guacamole-auth-jdbc-sqlserver/schema/001-create-schema.sql

```diff
@@ -0,0 +1,558 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/**
+ * List for permission data type.
+ */
+CREATE RULE [dbo].[guacamole_permission_list] 
+    AS
+    @list IN ('READ','UPDATE','DELETE','ADMINISTER')
+GO
+
+/**
+ * List for system permission data type.
+ */
+CREATE RULE [dbo].[guacamole_system_permission_list] 
+    AS
+    @list IN ('CREATE_CONNECTION',
+        'CREATE_CONNECTION_GROUP',
+        'CREATE_SHARING_PROFILE',
+        'CREATE_USER',
+        'ADMINISTER')
+GO
+
+/**
+ * The permission data type.
+ */
+CREATE TYPE [dbo].[guacamole_permission] FROM [nvarchar](10) NOT NULL
+
+/**
+ * The system permission data type.
+ */
+CREATE TYPE [dbo].[guacamole_system_permission] FROM [nvarchar](32) NOT NULL
+GO
+
+/**
+ * The connection_group table stores organizational and balancing groups.
+ */
+SET ANSI_NULLS ON
+SET QUOTED_IDENTIFIER ON

```

Comment:
>     SET ANSI_NULLS ON
>     SET QUOTED_IDENTIFIER ON

I see this quite a lot here. Why is this incantation necessary?

Reply:
So, most of the actual text here, minus the formatting, came from using SQL Server Management Studio to generate the output of the database minus any actual data.  As such, I took the file they spit out, re-arranged it, styled it to match Guacamole style, and that's what the .sql files you see here are.  I'm not entirely certain why SSMS found it necessary to spit this out so frequently, but I think that at least some of those options are only good in the current batch (up until the next "GO" statement), and I did run into issues trying to batch too many things together (got messages from SQL Server that certain queries had to be in their own batch).